openssh/USAGE.md

```diff
@@ -0,0 +1,128 @@
+
+One use case of quantum-safe ssh running in docker could be accessing a remote system without messing with its ssh(d) installation or other parts of the system you maybe don't want to interact with. This means minimal intrusion and everything can easily be removed again. This is done by running this docker image on said system and sharing its network space. Thus it is possible to access host ports from **within** the docker container. Normally, the use case of docker is one of isolation with some shared directories and published ports at max. So this solution works around the usual docker limitations.
+
+Additionally, it is advised to **change the default username and password** when building the image because your plan is to expose it to the world.
+
+```html
+Structure of quantum-safe remote access using docker containers
+
++------------------+                +----------------------+
+|      Client      |                |         Host         |
+|  +------------+  |                |  +----------------+  |
+|  |            |  |                |  |                |  |
+|  |   Docker   +--------------------->+     Docker     |  |
+|  |            |  |       Port 2222|  |                |  |
+|  +------------+  |                |  +-------+--------+  |
+|                  |                |          |           |
++------------------+                |  Port 22 v           |
+                                    |  +-------+--------+  |
+                                    |  |                |  |
+                                    |  |  sshd on host  |  |
+                                    |  |                |  |
+                                    |  +----------------+  |
+                                    |                      |
+                                    +----------------------+
+```
+### Set up the server (docker container on target-host)
+
+To start the ssh server, meaning the docker container on the host system, follow those instructions:
+- Build the docker image as usual, change username and password
+       
+       docker build --build-arg OQS_USER=<my-desired-username> --build-arg OQS_PASSWORD=<my-desired-password> -t oqs-openssh-img .
+- Run the docker image with
+
+       docker run -dit --network host --name oqs-ssh oqs-ssh-img
+
+- Or, if you want the container to automatically start with docker
+
+       docker run -dit --network host --name oqs-ssh --restart unless-stopped oqs-openssh-img
+
+The `--network host` option will attach the container directly to your host's network, sharing its IP. The sshd in the container is now accessible from the outside using the host's IP address and the specified port (2222 per default).
+
+Be aware that the port 2222 also needs to be open in any firewall's there may be!
+
+### Set up the client
+
+For the client side, you need to compile yourself a docker image as described in the 'Quick start' section of the [README.md](README.md). You then run it as normal with
```

Comment:
Why can't one use the (pre-existing/documented) docker image for this? The goal of this document is to be standing alone at hub.docker.com, just like e.g., https://hub.docker.com/r/openquantumsafe/nginx

Reply:
I was not aware of the full purpose of the USAGE.md, that it will be used directly on hub.docker.com. I will adjust all mentions of the image accordingly.